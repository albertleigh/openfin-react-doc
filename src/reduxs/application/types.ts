export interface IApplicationState {
    loading:boolean,
}
