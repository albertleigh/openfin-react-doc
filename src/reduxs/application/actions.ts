import { createAction, ActionFunctionAny, Action } from 'redux-actions';

export const APPLICATION_STARTED:string = 'APPLICATION_STARTED';
export const APPLICATION_READY:string = 'APPLICATION_READY';

export const applicationStarted:ActionFunctionAny<Action<void>> = createAction(APPLICATION_STARTED);
export const applicationReady:ActionFunctionAny<Action<void>> = createAction(APPLICATION_READY);
