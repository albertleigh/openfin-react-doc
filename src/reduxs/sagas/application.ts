import { delay } from 'redux-saga';
import { all, call, put, takeLatest } from 'redux-saga/effects';

import {
    APPLICATION_STARTED,
    applicationReady,
} from '..';

export function* handleApplicationLoading() {

    yield all([
        // you can add more sagas overhere
        call(delay,5000),
    ]);

    yield put.resolve(applicationReady());

}

export default function* () {
    yield takeLatest(APPLICATION_STARTED,handleApplicationLoading);
}