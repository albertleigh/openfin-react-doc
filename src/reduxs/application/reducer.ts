import { handleActions, Action } from 'redux-actions';

import {
    IApplicationState,
} from './types';

import { APPLICATION_READY } from './actions';

const defaultState:Partial<IApplicationState>={
    loading:true,
};

export default handleActions({
    [APPLICATION_READY]:(state,action)=>({
        ...state,
        loading:false,
    }),

},defaultState);