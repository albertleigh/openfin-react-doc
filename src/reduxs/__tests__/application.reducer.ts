import * as Actions from '../application/actions';
import reducer from '../application/reducer';

describe('Application reducer',()=>{

    it('APPLICATION_READY reduced correctly',()=>{
        const action:any = Actions.applicationReady();
        expect(reducer(undefined,action)).toMatchSnapshot();
    });

});