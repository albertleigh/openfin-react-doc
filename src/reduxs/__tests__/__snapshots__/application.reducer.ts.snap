// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application reducer APPLICATION_READY reduced correctly 1`] = `
Object {
  "loading": false,
}
`;
